### DISCUSSION TITLE
Streaming response in next.js 12.x.x, deployed on vercel

### DISCUSSION THREAD

**Discussion by vonsa** [vercel/next.js#49765] [Help]
### Summary

Hi All!

Does anyone know if it's possible to use streaming response using next.js 12.x.x and deploying on Vercel?

For example calling ```res.write()``` asynchronously a few times before calling ```res.end()```

With next.js 13 you seem to be able to use the edge runtime, but I moved back to version 12.3.4 because I encountered too many issues with version 13.

I've tried a lot of things and I have it working locally, but I'm having trouble getting it working on Vercel

**Comment by shawnesquivel** [comment]
Hey @vonsa 

There were some insights in this discussion but no concrete solution I saw #47076 

I'm also finding this issue with the SSE library. It works in local but not when deployed on Vercel

Does anybody have a working solution/repo they would like to share with the community?

Or perhaps some further context as to how you used Edge functions?

Library:
https://www.npmjs.com/package/sse

My frontend:
https://github.com/shawnesquivel/openai-javascript-course/blob/solutions/app/streaming/page.jsx

My route:
https://github.com/shawnesquivel/openai-javascript-course/blob/solutions/pages/api/streaming.js

**Comment by vonsa** [comment]
@shawnesquivel this is how I've solved it:

Updated to next 13.4.12, but not using the new app structure.

**In my api file:**

At the top of the file:
```
export const config = {
  runtime: "edge",
};
```

In my function:

I create a readable stream
```const clientStream = new ReadableStream(...)```

I return this
```return new Response(clientStream);```
